app/uk/gov/voa/valuetype/TypeName.scala

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.voa.valuetype
+
+trait TypeName {
```

Comment:
This seems a bit tricksy, could we get some meaningful comments and types??

Reply:
This is just code from the valuetype library that this service was using.  There is no Scala 2.12 version of it, so rather than try to rewrite all the code that uses it,  Steve Edmonds' recommendation was just to pull the code straight into the service.  He's already done that elsewhere.  I agree, this code seems pointless and unecessary, but I can't easily refactor at the moment.